sdk/core/azure-core/test/ut/CMakeLists.txt

```diff
@@ -47,6 +47,9 @@ add_executable (
      uuid.cpp
      )
 
+# Adding private headers from CORE to the tests so we can test the private APIs with no relative paths include.
+target_include_directories (${TARGET_NAME} PRIVATE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../../src> $<INSTALL_INTERFACE:include/az_core>)
```

Comment:
Interesting approach.

Are the `curl` headers the only ones that needed to be updated to use `_private` in azure::core and azure::identity?

Reply:
yes.
There are not private headers elsewhere.  Why did you mention identity?

Reply:
I understand the `$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../../src>`, but what does `$<INSTALL_INTERFACE:include/az_core>)` here do?

Reply:
@Jinming-Hu , that's for installation purposes (https://cmake.org/cmake/help/v3.13/command/target_include_directories.html)

I copy-pasted from Azure Core (wich was copy-pasted from the C repo).  It needs to be fixed for Azure Core. I think @antkmsft  will take care of it for vcpkg.

Since the tests are not expected to be installed, I can remove this second parameter here. Thanks!